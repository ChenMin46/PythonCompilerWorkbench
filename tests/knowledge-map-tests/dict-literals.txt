{}
{"Joe": 23}
{"Joe": 23, "Carol": 19, "Bobby": 32}
{"Joe": 23, 1000: 19, (1, 2): 32}
{"Joe": a + b}

# multiline
{"Joe": joe.get_age(),
 "Carol": carol.get_age(),
 "Bob": bob.get_age()}
