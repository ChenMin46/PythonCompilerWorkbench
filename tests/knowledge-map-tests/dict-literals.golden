{}
{"Joe": 23}
{"Joe": 23, "Carol": 19, "Bobby": 32}
{"Joe": 23, 1000: 19, (1, 2): 32}
{"Joe": a + b}

# multiline
{"Joe": joe.get_age(),
 "Carol": carol.get_age(),
 "Bob": bob.get_age()}
===
Module
  body: [
    Expr
      value: Dict >> '{}'
        keys: [
        ]
        values: [
        ]
    Expr
      value: Dict >> '{"Joe": 23}'
        keys: [
          Str >> '"Joe"'
            s: 'Joe'
        ]
        values: [
          Num >> '23'
            n: 23
        ]
    Expr
      value: Dict >> '{"Joe": 23, "Carol": 19, "Bobby": 32}'
        keys: [
          Str >> '"Joe"'
            s: 'Joe'
          Str >> '"Carol"'
            s: 'Carol'
          Str >> '"Bobby"'
            s: 'Bobby'
        ]
        values: [
          Num >> '23'
            n: 23
          Num >> '19'
            n: 19
          Num >> '32'
            n: 32
        ]
    Expr
      value: Dict >> '{"Joe": 23, 1000: 19, (1, 2): 32}'
        keys: [
          Str >> '"Joe"'
            s: 'Joe'
          Num >> '1000'
            n: 1000
          Tuple >> '(1, 2)'
            elts: [
              Num >> '1'
                n: 1
              Num >> '2'
                n: 2
            ]
            ctx: Load
        ]
        values: [
          Num >> '23'
            n: 23
          Num >> '19'
            n: 19
          Num >> '32'
            n: 32
        ]
    Expr
      value: Dict >> '{"Joe": a + b}'
        keys: [
          Str >> '"Joe"'
            s: 'Joe'
        ]
        values: [
          BinOp >> 'a + b'
            left: Name >> 'a'
              id: 'a'
              ctx: Load
            op: Add
            right: Name >> 'b'
              id: 'b'
              ctx: Load
        ]
    Expr
      value: Dict ML>> '{"Joe": joe.get_age(),\n "Carol": carol.get_age(),\n "Bob": bob.get_age()}'
        keys: [
          Str >> '"Joe"'
            s: 'Joe'
          Str >> '"Carol"'
            s: 'Carol'
          Str >> '"Bob"'
            s: 'Bob'
        ]
        values: [
          Call >> 'joe.get_age()'
            func: Attribute >> 'joe.get_age'
              value: Name >> 'joe'
                id: 'joe'
                ctx: Load
              attr: 'get_age'
              ctx: Load
            args: [
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
          Call >> 'carol.get_age()'
            func: Attribute >> 'carol.get_age'
              value: Name >> 'carol'
                id: 'carol'
                ctx: Load
              attr: 'get_age'
              ctx: Load
            args: [
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
          Call >> 'bob.get_age()'
            func: Attribute >> 'bob.get_age'
              value: Name >> 'bob'
                id: 'bob'
                ctx: Load
              attr: 'get_age'
              ctx: Load
            args: [
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
        ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Dict", 
              "value": "{}"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "{", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Joe\""
                }, 
                {
                  "id": "tid_1", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "23"
                }, 
                "}"
              ], 
              "id": "id_0", 
              "name": "Dict"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "{", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Joe\""
                }, 
                {
                  "id": "tid_2", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "23"
                }, 
                {
                  "id": "tid_3", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Carol\""
                }, 
                {
                  "id": "tid_4", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "19"
                }, 
                {
                  "id": "tid_5", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Bobby\""
                }, 
                {
                  "id": "tid_6", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "32"
                }, 
                "}"
              ], 
              "id": "id_0", 
              "name": "Dict"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "{", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Joe\""
                }, 
                {
                  "id": "tid_7", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "23"
                }, 
                {
                  "id": "tid_8", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "1000"
                }, 
                {
                  "id": "tid_9", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "19"
                }, 
                {
                  "id": "tid_10", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "(", 
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "1"
                    }, 
                    {
                      "id": "tid_11", 
                      "name": "token", 
                      "value": ","
                    }, 
                    " ", 
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "2"
                    }, 
                    ")"
                  ], 
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Tuple"
                }, 
                {
                  "id": "tid_12", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "32"
                }, 
                "}"
              ], 
              "id": "id_0", 
              "name": "Dict"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "{", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Joe\""
                }, 
                {
                  "id": "tid_13", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "a"
                    }, 
                    " ", 
                    {
                      "id": "tid_14", 
                      "name": "token", 
                      "value": "+"
                    }, 
                    " ", 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "b"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp", 
                  "op": "Add"
                }, 
                "}"
              ], 
              "id": "id_0", 
              "name": "Dict"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n\n", 
        {
          "id": "tid_15", 
          "name": "comment", 
          "value": "# multiline"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "{", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Joe\""
                }, 
                {
                  "id": "tid_16", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "attr": "get_age", 
                      "contents": [
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "joe"
                        }, 
                        ".get_age"
                      ], 
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Attribute"
                    }, 
                    "()"
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }, 
                {
                  "id": "tid_17", 
                  "name": "token", 
                  "value": ","
                }, 
                "\n ", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Carol\""
                }, 
                {
                  "id": "tid_18", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "attr": "get_age", 
                      "contents": [
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "carol"
                        }, 
                        ".get_age"
                      ], 
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Attribute"
                    }, 
                    "()"
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }, 
                {
                  "id": "tid_19", 
                  "name": "token", 
                  "value": ","
                }, 
                "\n ", 
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "\"Bob\""
                }, 
                {
                  "id": "tid_20", 
                  "name": "token", 
                  "value": ":"
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "attr": "get_age", 
                      "contents": [
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "bob"
                        }, 
                        ".get_age"
                      ], 
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Attribute"
                    }, 
                    "()"
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }, 
                "}"
              ], 
              "id": "id_0", 
              "name": "Dict"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
